server/server.go

```diff
@@ -723,15 +724,17 @@ func killConn(conn *clientConn) {
 	// we need to CANCEL the matching jobID first.
 	if sessVars.StmtCtx.IsDDLJobInQueue {
 		jobID := sessVars.StmtCtx.DDLJobID
-		err := kv.RunInNewTxn(context.Background(), conn.ctx.GetStore(), true, func(ctx context.Context, txn kv.Transaction) error {
-			// errs is the error per job, there is only one submitted
-			// err is the error of the overall task
-			errs, err := ddl.CancelJobs(txn, []int64{jobID})
+		se, err := session.CreateSession(conn.ctx.GetStore())
+		if err == nil {
+			var errs []error
+			se.GetSessionVars().CommonGlobalLoaded = true
+			se.GetSessionVars().InRestrictedSQL = true
+			se.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
```

Comment:
Maybe we can put this to `newSession`.  Then we can move this logic in  `sessionPool.get` to `newSession` too.

Reply:
it is a good idea to put 
```
se.GetSessionVars().CommonGlobalLoaded = true
se.GetSessionVars().InRestrictedSQL = true
```
in `newSession`, but for `DiskFullOpt_AllowedOnAlmostFull` I think it better not now.

Reply:
actually, if we put it in `newSession` it implies the `newSession` must be called, and we can not guarantee it. I think it ok to let the caller decide if there should have this code block

Reply:
`CancelJobs` function calls `newSession`, maybe we can remove them? 
Or, we needn't add these two lines to `newSession`.

Reply:
I'd like to remove these two line here.

Reply:
updated